### PR TITLE
Add new options to project resource

### PR DESCRIPTION
**Pull request by bilby91**
New options:
  - merge_method
  - only_allow_merge_if_pipeline_succeeds
  - only_allow_merge_if_all_discussions_are_resolved

### CODE REVIEW

**Comment by bilby91** [comment]
Is there any release process in place for this project ?

**Comment by roidelapluie** [comment]
Yes but we need to wait for hashicorp end of christmas freeze

**Comment by bilby91** [comment]
@mcanevet Can you provide more information around your tf files please ? Just by looking at the error I don't see a straight relation with `merge_method`.

File: gitlab/resource_gitlab_project.go [hunk 1]
```diff
@@ -190,6 +210,18 @@ func resourceGitlabProjectUpdate(d *schema.ResourceData, meta interface{}) error
 		options.Visibility = stringToVisibilityLevel(d.Get("visibility_level").(string))
 	}
 
+	if d.HasChange("merge_method") {
+		options.Visibility = stringToVisibilityLevel(d.Get("merge_method").(string))
```
Comment (pdecat) [inline gitlab/resource_gitlab_project.go]:
```suggestion
		options.MergeMethod = stringToMergeMethod(d.Get("merge_method").(string))
```

**Comment by roidelapluie** [comment]
Any comment on @mcanevet issue?

File: gitlab/resource_gitlab_project.go (hunk 1 above)
Comment (pdecat) [reply gitlab/resource_gitlab_project.go]: @roidelapluie this should resolve @mcanevet issue.

**Comment by mcanevet** [comment]
@pdecat your suggestion works. @bilby91 could you please add it to your PR?

**Comment by bilby91** [comment]
@mcanevet I just realized I didn't update the docs folders. Want me to get that done before merging ?

**Comment by mcanevet** [comment]
@bilby91 I guess @roidelapluie will ask for it. So if you have some spare time to do it, then please do it.

**Comment by roidelapluie** [comment]
Yes please

**Comment by bilby91** [comment]
Ping @roidelapluie

File: gitlab/resource_gitlab_project_test.go [hunk 2]
```diff
@@ -116,14 +131,17 @@ func TestAccGitlabProject_basic(t *testing.T) {
...
+						Path:                             fmt.Sprintf("foo.%d", rInt),
+						Description:                      "Terraform acceptance tests",
+						IssuesEnabled:                    true,
+						MergeRequestsEnabled:             true,
+						WikiEnabled:                      true,
+						SnippetsEnabled:                  true,
+						Visibility:                       gitlab.PublicVisibility,
+						MergeMethod:                      gitlab.FastForwardMerge,
+						OnlyAllowMergeIfPipelineSucceeds: true,
+						OnlyAllowMergeIfAllDiscussionsAreResolved: true,
```
Comment (roidelapluie) [inline gitlab/resource_gitlab_project_test.go]: did you run go fmt?

File: gitlab/util.go [hunk 3]
```diff
@@ -70,6 +71,19 @@ func stringToVisibilityLevel(s string) *gitlab.VisibilityValue {
 	}
 	return &value
 }
```
Comment (roidelapluie) [inline gitlab/util.go]: missing whiteline

Comment (bilby91) [reply gitlab/util.go]: I think my VSCode tries to do something different than `make fmt`. I pushed code again and CI is passing. I'm missing something ?

Comment (roidelapluie) [reply gitlab/util.go]: well you should run go fmt + add a blank line here :)

Comment (bilby91) [reply gitlab/util.go]: I think we should be good now :)

**Comment by roidelapluie** [comment]
Thanks!!

**Pull request merged**